src/components/Hoverable/index.js

```diff
@@ -107,6 +107,12 @@ class Hoverable extends Component {
                 }
             },
             onBlur: (el) => {
+                // Check if the blur event occurred due to clicking outside the element
+                // and the wrapperView contains the element that caused the blur and return early
+                if (el.currentTarget !== el.target && this.wrapperView.contains(el.target)) {
+                    return;
+                }
+
```

Comment:
We shouldn't return early here. The below `child.props.onBlur(el);` should always get called not to lose the event. Instead rewrite the conditions to when to call `this.setIsHovered(false);`

BTW, is the `el.currentTarget !== el.target` check required here?
Wouldn't this work too?

```js
if (!this.wrapperView.contains(el.target) && !this.wrapperView.contains(el.relatedTarget)) {
    this.setIsHovered(false);
}

```

Reply:
@getusha Can you please have a look?

Reply:
@getusha Can you please update screenshots as well.